src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,174 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * The TaskList class stores the list of items. Items are stored in
+ * an ArrayList of Strings. Supports adding of new items to the list
+ * and printing of the entire current list.
+ */
+public class TaskList {
+    /** The list of tasks stored */
+    private ArrayList<Task> list;
+    private Ui ui;
+
+    /**
+     * Constructor for TaskList class.
+     */
+    public TaskList() {
+        this.list = new ArrayList<Task>();
+        this.ui = new Ui();
+    }
+
+    /**
+     * Constructor for TaskList.
+     * Constructs the list based on the given array of strings.
+     *
+     * @param tasklist An array of strings of tasks.
+     */
+    public TaskList(ArrayList<String> tasklist) {
+        this();
+        for (String s : tasklist) {
+            String[] splitData = s.split(",");
+            switch(splitData[0]) {
+            case("T"):
+                addTask(splitData[2], Boolean.parseBoolean(splitData[1]));
+                break;
+            case("D"):
+                addTask(splitData[2], LocalDate.parse(splitData[3]),"deadline", Boolean.parseBoolean(splitData[1]));
+                break;
+            case("E"):
+                addTask(splitData[2], LocalDate.parse(splitData[3]), "event", Boolean.parseBoolean(splitData[1]));
+            }
+        }
+    }
+
+    /**
+     * Returns the array of tasks.
+     *
+     * @return Array of tasks.
+     */
+    public ArrayList<Task> getList() {
+        return this.list;
+    }
+
+    /**
+     * Adds a task the list and saves to the file.
+     *
+     * @param description The description of the task.
+     */
+    public String addTask(String description) {
```

Comment:
Is there a reason why you decided to name these methods all addTask rather than addToDo, addDeadline, addEvent?